src/typeahead/typeahead-window.ts

```diff
@@ -73,12 +133,26 @@ export class NgbTypeaheadWindow implements OnInit {
   @Input() resultTemplate: TemplateRef<ResultTemplateContext>;
 
   /**
+   * A template used to display a no results message in the dropdown window
+   */
+  @Input() noResultsTemplate: TemplateRef<any>;
+
+  /**
+   * A template to override a matching result default display
+   */
+  @Input() windowTemplate: TemplateRef<WindowTemplateContext>;
+
+  /**
    * Event raised when user selects a particular result row
    */
   @Output('select') selectEvent = new EventEmitter();
 
   @Output('activeChange') activeChangeEvent = new EventEmitter();
 
+  _getWindowContext() { return this._context; }
+
+
+  _getNoResultsContext() { return {term: this.term}; }
```

Comment:
I'm unsure of why this method (and the one above it) is prefixed with an underscore. Is it because the method is a getter for a private member variable?

Reply:
I believe this was the pattern already established in the file to indicate the private function which is used by the template to get the context for the two different ng-template sections. The contexts are not intended to be exposed outside of the control.